src/common/mfu_daos.c

```diff
@@ -109,48 +103,73 @@ static void mfu_daos_copy_stats_print(mfu_daos_copy_stats_t* stats)
     const char* read_size_units;
     mfu_format_bytes(stats->bytes_read, &read_size_tmp, &read_size_units);
 
-    /* compute read rate */
-    double read_rate = 0;
-    if (stats->bytes_read > 0) {
-        read_rate = (double) stats->bytes_read / rel_time;
-    }
-
-    /* convert read rate to units */
-    double read_rate_tmp;
-    const char* read_rate_units;
-    mfu_format_bw(read_rate, &read_rate_tmp, &read_rate_units);
-
     /* convert write size to units */
     double write_size_tmp;
     const char* write_size_units;
     mfu_format_bytes(stats->bytes_written, &write_size_tmp, &write_size_units);
 
-    /* compute write rate */
-    double write_rate = 0;
-    if (stats->bytes_written > 0) {
-        write_rate = (double) stats->bytes_written / rel_time;
-    }
-
-    /* convert write rate to units */
-    double write_rate_tmp;
-    const char* write_rate_units;
-    mfu_format_bw(write_rate, &write_rate_tmp, &write_rate_units);
-
-    MFU_LOG(MFU_LOG_INFO, "Started    : %s", starttime_str);
-    MFU_LOG(MFU_LOG_INFO, "Completed  : %s", endtime_str);
-    MFU_LOG(MFU_LOG_INFO, "Seconds    : %.3lf", rel_time);
-    MFU_LOG(MFU_LOG_INFO, "Objects    : %" PRId64, stats->total_oids);
-    MFU_LOG(MFU_LOG_INFO, "  D-Keys   : %" PRId64, stats->total_dkeys);
-    MFU_LOG(MFU_LOG_INFO, "  A-Keys   : %" PRId64, stats->total_akeys);
-
-    MFU_LOG(MFU_LOG_INFO, "Bytes read    : %.3lf %s (%" PRId64 " bytes)",
-            read_size_tmp, read_size_units, stats->bytes_read);
-    MFU_LOG(MFU_LOG_INFO, "Bytes written : %.3lf %s (%" PRId64 " bytes)",
-            write_size_tmp, write_size_units, stats->bytes_written);
-    MFU_LOG(MFU_LOG_INFO, "Read rate  : %.3lf %s",
-            read_rate_tmp, read_rate_units);
-    MFU_LOG(MFU_LOG_INFO, "Write rate : %.3lf %s",
-            write_rate_tmp, write_rate_units);
+    MFU_LOG(MFU_LOG_INFO, "Started       : %s", starttime_str);
+    MFU_LOG(MFU_LOG_INFO, "Completed     : %s", endtime_str);
+    MFU_LOG(MFU_LOG_INFO, "Seconds       : %.3lf", rel_time);
+    MFU_LOG(MFU_LOG_INFO, "Objects       : %" PRId64, stats->total_oids);
+    MFU_LOG(MFU_LOG_INFO, "  D-Keys      : %" PRId64, stats->total_dkeys);
+    MFU_LOG(MFU_LOG_INFO, "  A-Keys      : %" PRId64, stats->total_akeys);
+
+    if (print_read) {
+        MFU_LOG(MFU_LOG_INFO, "Bytes read    : %.3lf %s (%" PRId64 " bytes)",
+                read_size_tmp, read_size_units, stats->bytes_read);
+    }
+    if (print_write) {
+        MFU_LOG(MFU_LOG_INFO, "Bytes written : %.3lf %s (%" PRId64 " bytes)",
+                write_size_tmp, write_size_units, stats->bytes_written);
+    }
+
+    if (print_read_rate) {
+        /* Compute read rate */
+        double read_rate = 0;
+        if (stats->bytes_read > 0) {
+            read_rate = (double) stats->bytes_read / rel_time;
+        }
```

Comment:
It's not likely, but I always get paranoid about a possible divide by zero in places like this.  To be safe, you can add in a ``rel_time > 0`` check.

Reply:
Oh, that's actually what I _**meant**_ to do to begin with:
```
if (rel_tim > 0) {
    read_rate = (double) stats->bytes_read / rel_time;
}
```
Since it doesn't matter if `bytes_read` if 0. (that would just be 0)
Thanks for noticing this!